Assets/Scripts/SS3D/Systems/Inventory/Items/ItemActor.cs

```diff
@@ -124,17 +119,31 @@ protected override void OnStart()
             }
 
             // Items can't have no size
-            if (_size.x == 0)
+            if (item.Size.x == 0)
             {
-                _size = new Vector2Int(1, _size.y);
+                item.Size = new Vector2Int(1, item.Size.y);
             }
 
-            if (_size.y == 0)
+            if (item.Size.y == 0)
             {
-                _size = new Vector2Int(_size.x, 1);
+                item.Size = new Vector2Int(item.Size.x, 1);
             }
         }
 
+        /// <summary>
+        /// Changes the item actor's item id
+        /// </summary>
+        /// <param name="id">AssetDatabase's ItemId</param>
+        public void SetId(ItemId id)
```

Comment:
why do we need this ? is that new or was it something in item before ?

Reply:
It i just set the id manually it throws a null ref exception, because the AssetDatabase will create an ItemActor and try to access Item before the Awake event is called

Reply:
Okay cool